cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sinks/StreamSink.java

```diff
@@ -0,0 +1,93 @@
+import co.cask.cdap.templates.etl.api.Property;
+import co.cask.cdap.templates.etl.api.StageConfigurer;
+import co.cask.cdap.templates.etl.api.config.ETLStage;
+import co.cask.cdap.templates.etl.api.realtime.RealtimeSink;
+import co.cask.cdap.templates.etl.api.realtime.SinkContext;
+import co.cask.cdap.templates.etl.common.Properties;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+/**
+ * Real-time sink for Streams
+ */
+public class StreamSink extends RealtimeSink<StructuredRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(StreamSink.class);
+
+  @Override
+  public void configure(StageConfigurer configurer) {
+    configurer.setDescription("Real-time sink that outputs to the specified stream.");
+    configurer.addProperty(new Property(Properties.Stream.NAME,
+                                        "The name of the stream to output to . Must be a valid stream name. " +
+                                          "The stream will be created if it does not exist",
+                                        true));
+    configurer.addProperty(new Property(Properties.Stream.DATA_FIELD,
+                                        "Field name in the generic record that contains the data to be written to " +
+                                          "the specified stream. The data could be either in avro/binary format or " +
+                                          "stream event byte buffer format. If unspecified, 'data' key is used and " +
+                                          "avro/binary format is assumed to be default.",
+                                        false));
+  }
+
+  @Override
+  public void configurePipeline(ETLStage stageConfig, PipelineConfigurer pipelineConfigurer) {
+    Map<String, String> properties = stageConfig.getProperties();
+    String streamName = properties.get(Properties.Stream.NAME);
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(streamName), "Stream name should be non-null, non-empty.");
+    pipelineConfigurer.addStream(new Stream(streamName));
+  }
+
+  @Override
+  public void write(Iterable<StructuredRecord> structuredRecords) throws Exception {
+    SinkContext context = getContext();
+    Map<String, String> runtimeArguments = context.getRuntimeArguments();
+    String streamName = runtimeArguments.get(Properties.Stream.NAME);
+    String dataField = runtimeArguments.containsKey(Properties.Stream.DATA_FIELD) ?
+      runtimeArguments.get(Properties.Stream.DATA_FIELD) : Properties.Stream.DEFAULT_DATA_FIELD;
+
+    for (StructuredRecord structuredRecord : structuredRecords) {
+      Object data = structuredRecord.get(dataField);
+      if (data == null) {
+        LOG.debug("Found null data. Skipping record.");
+        return;
+      }
+      if (data.getClass().isAssignableFrom(StreamEvent.class)) {
```

Comment:
i would assume one record translates into one stream event.  The event body would come from one field in the record.  Headers could come from another field, as it's possible to have a map<string,string> field in a record.